### PR TITLE
feat: Add gitlab group visibility to group entity via annotation

### LINKED ISSUE #22887: 🚀 Feature: GitlabOrg-Provider should annotate Groups with group visibility information
### 🔖 Feature description

I think it would be a good idea to write the project visibility level of groups created by the gitlab org provider with their visibility level. This information is lost right now - but it I think it would be good for things like rbac rules if we would ingest them. 

### 🎤 Context

At the moment we have no chance to distinguish groups and their visibility level and use this information in permissions

### ✌️ Possible Implementation

extend the gitlab group type with visibility_level and write it to the annotations of the group entity (like the provider does for the team-path). 

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by markussiebert**
## Hey, I just made a Pull Request!

This PR adds an annotation to groups ingested by gitlab org provider and annotates them with their visibility level:

fixes #22887

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by tudi2d — CHANGES_REQUESTED** [review]
Thanks for your contribution - heads-up that requiring the `visibility` annotation currently breaks the types. Is there a chance to make it optional? Otherwise it seems reasonable & let's get some 2nd eyes on it to ship it:)

File: plugins/catalog-backend-module-gitlab/src/lib/types.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export type GitLabGroup = {
   name: string;
   full_path: string;
   description?: string;
+  visibility: string;
```
[the remark this answers is not in the document]
Comment (markussiebert) [reply plugins/catalog-backend-module-gitlab/src/lib/types.ts:79]: I extended the nested group graphql call, so I think there should be no further problem

**Comment by markussiebert** [comment]
Not stale :-)

**Comment by benjdlambert** [comment]
Anyone that's consuming the `GitLabGroup` type that's exported is now going to have to provide `visibility` in that now, which would be a breaking change. Not sure what the impact of that is though, maybe it's fine to just accept that this is a breaking change,  otherwise @tudi2d is correct, that we should make this optional instead, even though the producer of this would always make sure its set.

**Review by benjdlambert — DISMISSED** [review]

**Comment by markussiebert** [comment]
I changed the visibility to optional - you are right, it is totally fine to make it optional and set it all the time

**Review by mikeyhc — APPROVED** [review]

**Review by tudi2d — APPROVED** [review]
thanks for adjusting to the review. looks good from my side ✅

**Review by benjdlambert — APPROVED** [review]
Nice thanks! :tada:

**Pull request merged**